public/app/features/query/state/DashboardQueryRunner/SnapshotWorker.ts

```diff
@@ -0,0 +1,42 @@
+import { Observable, of } from 'rxjs';
+import { cloneDeep } from 'lodash';
+import { AnnotationEvent } from '@grafana/data';
+
+import { DashboardQueryRunnerOptions, DashboardQueryRunnerWorker, DashboardQueryRunnerWorkerResult } from './types';
+import { emptyResult } from './utils';
+
+export class SnapshotWorker implements DashboardQueryRunnerWorker {
+  canWork({ dashboard }: DashboardQueryRunnerOptions): boolean {
+    const snapshots = dashboard.annotations.list.find((a) => a.enable && Boolean(a.snapshotData));
+    return Boolean(snapshots);
```

Comment:
Could do the following to avoid `Boolean(snapshots)`, if you wanted:
```suggestion
    return dashboard.annotations.list.some((a) => a.enable && Boolean(a.snapshotData));
```

Reply:
Excellent suggestion, out of curiosity, which is the most performant `some` or `find`.

Reply:
Did some very unscientific benchmarks and it seems like they're basically the same in terms of performance

Reply:
Thank you, and I think `some` adds better readability.

Reply:
Fixed